src/server/cell_labeling_app/main.py

```diff
@@ -9,52 +13,121 @@
 from cell_labeling_app.user_authentication.user_authentication import login
 
 
-def create_app(config_file: Path, debug=False):
-    if not config_file.exists():
-        raise ValueError('Config file does not exist')
-    if not config_file.suffix == '.py':
-        raise ValueError('Config file must be a python module ending in '
-                           '".py"')
+class _BackupSchema(argschema.ArgSchema):
+    frequency = argschema.fields.Integer(
+        default=60 * 5,
+        description='Number of seconds to wait before creating a new backup'
+    )
 
-    template_dir = (Path(__file__).parent.parent.parent / 'client').resolve()
-    static_dir = template_dir
-    app = Flask(__name__, static_folder=static_dir,
-                          template_folder=str(template_dir))
-    app.register_blueprint(api)
-    app.config.from_pyfile(filename=str(config_file))
-    db.init_app(app)
-    app.register_blueprint(users)
-    app.secret_key = app.config['SESSION_SECRET_KEY']
 
-    log_level = logging.DEBUG if debug else logging.INFO
-    logging.basicConfig(filename=app.config['LOG_FILE'], level=log_level)
+class AppSchema(argschema.ArgSchema):
+    database_path = argschema.fields.InputFile(
+        required=True,
+        description='Database path',
+    )
+    ARTIFACT_DIR = argschema.fields.InputDir(
+        required=True,
+        description='Path to h5 files storing data required for the '
+                    'app to run'
+    )
+    PREDICTIONS_DIR = argschema.fields.InputDir(
+        required=True,
+        description='Path to pre-classification predictions'
+    )
+    PORT = argschema.fields.Integer(
+        default=5000,
+        description='Port the app should run on'
+    )
+    FIELD_OF_VIEW_DIMENSIONS = argschema.fields.Tuple(
+        (argschema.fields.Int, argschema.fields.Int),
+        default=(512, 512),
+        description='Field of view dimensions'
+    )
+    LOG_FILE = argschema.fields.OutputFile(
+        required=True,
+        description='Path to where logs should be written'
+    )
+    LABELS_PER_REGION_LIMIT = argschema.fields.Integer(
```

Comment:
What's the behavior we'll end up using for the "real" labeling effort? Do we want to limit to a hard value of 3 or maybe downweight the likelihood of drawing a region based on how many times it's been labeled? I suppose a hard stop at three "labelings" is the easiest way to do this and the downweighting would only really make sense if we were rich with labelers.

Reply:
I was expecting to use this default of 3, unless you feel strongly otherwise

Reply:
I personally feel that 1 would be sufficient but some on the science team thought otherwise. We would get more labels quicker this way, and on average we would be training on expert opinion.

Reply:
I don't. Like I said, we're likely not labeler rich so it makes sense to go for the 3 at most.